packages/frontend/src/hooks/cartesianChartConfig/useCartesianChartConfig.ts

```diff
@@ -120,21 +122,40 @@ const useCartesianChartConfig = ({
     stacking,
     cartesianType,
 }: Args) => {
-    // FIXME: this might not be necessary
-    const hasInitialValue =
-        !!initialChartConfig &&
-        isCompleteLayout(initialChartConfig.layout) &&
-        isCompleteEchartsConfig(initialChartConfig.eChartsConfig);
-    const [dirtyLayout, setDirtyLayout] = useState<
-        Partial<CartesianChart['layout']> | undefined
-    >(initialChartConfig?.layout);
-    const [dirtyEchartsConfig, setDirtyEchartsConfig] = useState<
-        Partial<CartesianChart['eChartsConfig']> | undefined
-    >(initialChartConfig?.eChartsConfig);
-    const isInitiallyStacked = (dirtyEchartsConfig?.series || []).some(
-        (series: Series) => series.stack !== undefined,
-    );
+    const isInitiallyStacked = (
+        initialChartConfig?.eChartsConfig.series || []
+    ).some((series: Series) => series.stack !== undefined);
     const [isStacked, setIsStacked] = useState<boolean>(isInitiallyStacked);
+    const prevInitialChartConfig = usePrevious(initialChartConfig);
+    const [dirtyLayout, setDirtyLayout] = useState(initialChartConfig?.layout);
+    const [dirtyEchartsConfig, setDirtyEchartsConfig] = useState(
+        initialChartConfig?.eChartsConfig,
+    );
+
+    const validConfig: CartesianChart = useMemo(() => {
+        return isCompleteLayout(dirtyLayout) &&
+            isCompleteEchartsConfig(dirtyEchartsConfig)
+            ? {
+                  layout: dirtyLayout,
+                  eChartsConfig: dirtyEchartsConfig,
+              }
+            : EMPTY_CARTESIAN_CHART_CONFIG;
+    }, [dirtyLayout, dirtyEchartsConfig]);
+
+    const hasChartConfigChangedExternally = useMemo(() => {
+        return (
+            !isEqual(initialChartConfig, validConfig) &&
+            !isEqual(initialChartConfig, prevInitialChartConfig)
+        );
+    }, [initialChartConfig, prevInitialChartConfig, validConfig]);
```

Comment:
The only reason this is duplicated in `packages/frontend/src/components/LightdashVisualization/VisualizationConfigCartesian.tsx` is so that we keep the call to update the state in the explore there and don't have to move it here